src/expr/impl/src/scalar/vector.rs

```diff
@@ -576,3 +576,107 @@ fn l2_normalize(vector: VectorRef<'_>) -> Result<VectorVal> {
         .map_err(|_| ExprError::NumericOverflow)?;
     Ok(result)
 }
+
+/// ```slt
+/// query R
+/// SELECT vector_dims('[1,2,3]'::vector(3));
+/// ----
+/// 3
+///
+/// query R
+/// SELECT vector_dims('[10,20,30,40]'::vector(4));
+/// ----
+/// 4
+///
+/// query error dimensions
+/// SELECT vector_dims('[]'::vector(0));
+/// ```
+#[function("vector_dims(vector) -> int4")]
+fn vector_dim(v: VectorRef<'_>) -> Result<i32> {
```

Comment:
The dimension of vector is inherent in its known data type. Is it necessary to provide a function to get from the vector?